contracts/instance/service/PricingService.sol

```diff
@@ -0,0 +1,275 @@
+                );
+            }
+
+            if ((premium.distributionFeeVarAmount) != (premium.discountAmount + premium.distributionOwnerFeeVarAmount + premium.commissionAmount)) {
+                revert ErrorIPricingServiceFeeCalculationMismatch(
+                    premium.distributionFeeFixAmount,
+                    premium.distributionFeeVarAmount,
+                    premium.distributionOwnerFeeFixAmount,
+                    premium.distributionOwnerFeeVarAmount,
+                    premium.commissionAmount,
+                    premium.discountAmount
+                );
+            }
+        }
+    }
+
+    // internal functions
+    function _getFixedFeeAmounts(
+        uint256 netPremiumAmount,
+        ISetup.ProductSetupInfo memory productInfo,
+        IComponents.PoolInfo memory poolInfo,
+        ISetup.DistributionSetupInfo memory distInfo,
+        IBundle.BundleInfo memory bundleInfo
+    )
+        internal
+        view
+        returns (
+            IPolicy.Premium memory premium
+        )
+    {
+        // initial premium amount is the net premium
+        premium.netPremiumAmount = netPremiumAmount;
+        premium.fullPremiumAmount = netPremiumAmount;
+
+        uint256 t = productInfo.productFee.fixedFee;
+        premium.productFeeFixAmount = t;
+        premium.fullPremiumAmount += t;
+
+        t = poolInfo.poolFee.fixedFee;
+        premium.poolFeeFixAmount = t;
+        premium.fullPremiumAmount += t;
+
+        t = bundleInfo.fee.fixedFee;
+        premium.bundleFeeFixAmount = t;
+        premium.fullPremiumAmount += t;
+
+        t = distInfo.distributionFee.fixedFee;
+        premium.distributionFeeFixAmount = t;
+        premium.fullPremiumAmount += t;
+    }
+
+    function _calculateVariableFeeAmounts(
+        IPolicy.Premium memory premium,
+        ISetup.ProductSetupInfo memory productInfo,
+        IComponents.PoolInfo memory poolInfo,
+        ISetup.DistributionSetupInfo memory distInfo,
+        IBundle.BundleInfo memory bundleInfo
+    )
+        internal
+        view
+        returns (
+            IPolicy.Premium memory intermadiatePremium
+        )
+    {
+        UFixed netPremiumAmount = UFixedLib.toUFixed(premium.netPremiumAmount);
+
+        uint256 t = (netPremiumAmount * productInfo.productFee.fractionalFee).toInt();
+        premium.productFeeVarAmount = t;
+        premium.fullPremiumAmount += t;
+
+        t = (netPremiumAmount * poolInfo.poolFee.fractionalFee).toInt();
+        premium.poolFeeVarAmount = t;
+        premium.fullPremiumAmount += t;
+
+        t = (netPremiumAmount * bundleInfo.fee.fractionalFee).toInt();
+        premium.bundleFeeVarAmount = t;
+        premium.fullPremiumAmount += t;
+
+        t = (netPremiumAmount * distInfo.distributionFee.fractionalFee).toInt();
+        premium.distributionFeeVarAmount = t;
+        premium.fullPremiumAmount += t;
+
+        return premium;
+    }
+
+    function _calculateFeeAmount(
```

Comment:
We should rename this to make it clearer that it does (`calculateFeeAmount` was clear in the scope of the `DistributionService`)

```suggestion
    function _calculateDistributionFees(
```

Reply:
done